### PR TITLE
feat: defi section ledger open app gate

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

This PR implements Ledger over all DeFi section signing actions.

## Issue (if applicable)

closes https://github.com/shapeshift/web/issues/7579

## Risk

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

Low to none

## Testing

- Ensure all signing actions across the DeFi section (deposit, withdraw, claim) are properly guarded by the Ledger Open App gate, i.e pop up a modal to open the relevant Ledger app if no/wrong app is opened
- Test this over:
  - ETH/FOX (UNI-V2)
  - FOX Farming
  - FOXy
  - THORChain savers
  - Cosmos SDK

### Engineering

- ^

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

- ^

## Screenshots (if applicable)

- UNI-V2

Screenshot 2024-08-27 at 09 14 58
Screenshot 2024-08-27 at 09 14 41
Screenshot 2024-08-27 at 09 13 59
Screenshot 2024-08-27 at 09 13 50
Screenshot 2024-08-27 at 09 12 50
Screenshot 2024-08-27 at 09 12 29

- FOX Farming

Screenshot 2024-08-27 at 09 28 09
Screenshot 2024-08-27 at 09 27 39

- FOXy

Screenshot 2024-08-27 at 10 16 31
Screenshot 2024-08-27 at 10 07 03

- THORChain Savers

 
Screenshot 2024-08-27 at 10 19 34
Screenshot 2024-08-27 at 10 18 35

- Cosmos SDK

image
image
image

### CODE REVIEW

File: src/lib/utils/thorchain/hooks/useSendThorTx.tsx [hunk 1]
```diff
@@ -333,6 +335,8 @@ export const useSendThorTx = ({
             wallet,
           })
 
+          await checkLedgerAppOpenIfLedgerConnected(asset.chainId)
```
Comment (NeOMakinG) [inline src/lib/utils/thorchain/hooks/useSendThorTx.tsx:338]: preferably-blocking: seems like this line is not handling every cases (try to deposit into runepool for example, or maybe even THORChain BTC Savers)

Comment (gomesalexandre) [reply src/lib/utils/thorchain/hooks/useSendThorTx.tsx]: Nice catch ser, blaming jetlag on this one. 18ee336679

File: src/plugins/cosmos/hooks/useStakingAction/useStakingAction.tsx [hunk 2]
```diff
@@ -68,8 +70,10 @@ export const useStakingAction = () => {
       const memo = shapeshiftValidators.includes(validator) ? 'Delegated with ShapeShift' : ''
 
       const { accountNumber } = bip44Params
+
+      await checkLedgerAppOpenIfLedgerConnected(asset.chainId)
+      const address = await adapter.getAddress({ accountNumber, wallet })
```
Comment (NeOMakinG) [inline src/plugins/cosmos/hooks/useStakingAction/useStakingAction.tsx:75]: q: any reason why this line move? diff sanity

Comment (NeOMakinG) [inline src/plugins/cosmos/hooks/useStakingAction/useStakingAction.tsx:75]: 💜

**Review by NeOMakinG — APPROVED** [review]
Thorchain savers, Fox Farming, Univ2: https://jam.dev/c/82228eea-9378-4aa8-b55b-6b5ce1abce12

I can't test foxy on my side!

Cosmos:
image

Working like a charm

**Pull request merged**